examples/fix.kl

```diff
@@ -0,0 +1,43 @@
+#lang kernel
+
+(datatype (Fix F)
+  (in (F (Fix F))))
+
+(define out
+  (lambda (x)
+    (case x
+      ((in y) y))))
+
+(datatype (NatF A)
+  (zero)
+  (succ A))
+
+(datatype (ListF A L)
+  (nil)
+  (cons A L))
+
+(define length
+  (flet (length (xs)
+          (case (out xs)
+            [(nil) (in (zero))]
+            [(cons y ys) (in (succ (length ys)))]))
+    length))
+
+
+(define l1 (the (Fix (ListF String)) (in (cons "hello" (in (cons "world" (in (nil))))))))
```

Comment:
does the `the` imply that our type inference algorithm unable to infer `l1`'s type?

Reply:
It does not - that was there for readability. I just pushed a version without it to make sure it keeps working.